src/nekontrol/language.py

```diff
@@ -52,26 +57,21 @@ def __init__(
         self.tctx = tctx
 
     def prepare(self) -> Runnable:
-        raise NotImplementedError()
+        ...
 
     def cleanup(self):
         pass
 
-    @property
-    def kattis_name(self) -> str:
-        raise NotImplementedError()
-
     def __enter__(self) -> Runnable:
         return self.prepare()
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.cleanup()
 
 
-class InterpretedLanguage(Language):
-    @property
-    def bins(self) -> list[str]:
-        raise NotImplementedError()
+class InterpretedLanguage(Language, Protocol):
+    bins: ClassVar[list[str]]
+    bin: str
```

Comment:
What is this bin?

Reply:
i cant link but it is an instance variable defined a few lines down, so it needs to be declared for the protocol class.

Reply:
https://github.com/Quaqqer/nekontrol/blob/optional-tests/src/nekontrol/language.py#L87 here